internal/pkg/coordinator/monitor.go

```diff
@@ -453,6 +453,23 @@ func runCoordinatorOutput(ctx context.Context, cord Coordinator, bulker bulk.Bul
 }
 
 func runUnenroller(ctx context.Context, bulker bulk.Bulk, policyID string, unenrollTimeout time.Duration, l zerolog.Logger, checkInterval time.Duration, agentsIndex string) {
+	// When fleet-server is offline for a long period and finally recover, it means that the connected
+	// agent will be offline for a long period of time since their last checkin and fleet server will
+	// start unenrolling every Elastic Agent from the system. Instead on boot the Elastic Agent
+	// should wait at least the unenrolltimeout period before actively unenrolling agents in the system.
+	// This give a grace period to the Elastic Agent to connect back to the system.
+	l.Info().
+		Dur("checkInterval", checkInterval).
+		Dur("unenrollTimeout", unenrollTimeout).
+		Msg("giving a grace period to Elastic Agent before enforcing unenrollTimeout monitor")
+
+	select {
+	case <-time.After(unenrollTimeout):
```

Comment:
I'd change this to NewTimer, cause this select can be exited sooner on context.Done:
"The underlying Timer is not recovered by the garbage collector until the timer fires. If efficiency is a concern, use NewTimer instead and call Timer.Stop if the timer is no longer needed."

Reply:
Yeah, My first shot was using a timer, I've considered this would only be called for a few time n agent policy time. But I will change it.